File: src/Java/Programmers/Lv2/PR_연습문제_다음큰숫자.java
package Java.Programmers.Lv2;

class NextBigNum{
    public int nextBigNum(int n){
        int count = checkNum(n);

        for(int i = n + 1;;i++){
            if(checkNum(i) == count) return i;
        }
    }

    public int checkNum(int n){
        int count = 0;
        String binary = Integer.toBinaryString(n);
        for(int i = 0; i < binary.length(); i++){
            if(binary.charAt(i) == '1') count++;
        }

        return count;
    }

    public static void main(String[] args){
        int n = 78;
        NextBigNum sol = new NextBigNum();

        System.out.println(sol.nextBigNum(n));
    }
}

File: src/Kotlin/PR_연습문제_다음큰숫자.kt
package Kotlin

class NextBigNum{
    fun nextBigNum(num: Int): Int{
        val count: Int = checkNum(num)
        var answer: Int = num+1
        while(true){
            if(checkNum(answer) == count) break
            answer++
        }

        return answer
    }

    fun checkNum(num : Int): Int{
        var count:Int = 0
        val binary:String =Integer.toBinaryString(num)

        for(i in binary){
            if(i == '1') count++
        }

        return count
    }
}
fun main(){
    val num = 78
    val sol = NextBigNum()

    println(sol.nextBigNum(78))
}